src/Service/MailService.php

```diff
@@ -94,15 +95,38 @@ public function send(): ResultInterface
         }
 
         //save copy of sent message to folders
-        if ($this->mailOptions->getSaveSentMessageFolder()) {
+        if ($this->mailOptions->getTransport() == 'Laminas\Mail\Transport\Smtp' &&
+            $this->mailOptions->getSaveSentMessageFolder()) {
             foreach ($this->mailOptions->getSaveSentMessageFolder() as $folder) {
-                $this->storage->appendMessage($this->message->toString(), $folder);
+                $this->storage = $this->createStorage();
```

Comment:
If I understand correctly, `$this->storage` is reused for each folder.
If so, then it should not be called from inside the foreach loop (because it creates a new connection per iteration).

Reply:
good point
i'll make the change